#############
## SAMPLES ##
#############

#ID	run	condition	strand
#E15_KO_NT	E15_KO_NT_htseq	KO_NT	Unstranded
#E15_KO_P	E15_KO_P_htseq	KO_P	Unstranded
#E15_KO_R24	E15_KO_R24_htseq	KO_R24	Unstranded
#E15_KO_R3	E15_KO_R3_htseq	KO_R3	Unstranded
#E15_WT_NT	E15_WT_NT_htseq	WT_NT	Unstranded
#E15_WT_P	E15_WT_P_htseq	WT_P	Unstranded
#E15_WT_R24	E15_WT_R24_htseq	WT_R24	Unstranded
#E15_WT_R3	E15_WT_R3_htseq	WT_R3	Unstranded
#E29_KO_NT	E29_KO_NT_htseq	KO_NT	Unstranded
#E29_KO_P	E29_KO_P_htseq	KO_P	Unstranded
#E29_KO_R24	E29_KO_R24_htseq	KO_R24	Unstranded
#E29_KO_R3	E29_KO_R3_htseq	KO_R3	Unstranded
#E29_WT_NT	E29_WT_NT_htseq	WT_NT	Unstranded
#E29_WT_P	E29_WT_P_htseq	WT_P	Unstranded
#E29_WT_R24	E29_WT_R24_htseq	WT_R24	Unstranded
#E29_WT_R3	E29_WT_R3_htseq	WT_R3	Unstranded
#E31_KO_NT	E31_KO_NT_htseq	KO_NT	Unstranded
#E31_KO_P	E31_KO_P_htseq	KO_P	Unstranded
#E31_KO_R24	E31_KO_R24_htseq	KO_R24	Unstranded
#E31_KO_R3	E31_KO_R3_htseq	KO_R3	Unstranded
#E31_WT_NT	E31_WT_NT_htseq	WT_NT	Unstranded

# Conda environments:
# STRAND  --> how_are_we_stranded_here  v1.0.1
# HTSEQ   --> htseq-count  v0.13.5
# STAR    --> STAR  v2.7.9a

####################
## Create Folders ##
####################

#mkdir 0.MISCELLANEOUS
#mkdir RESULTS_STAR
#mkdir RESULTS_STAR/NO_FILTERED
#mkdir RESULTS_STAR/FILTERED
#mkdir RESULTS_HTSEQ
#mkdir RESULTS_HTSEQ/NO_FILTERED
#mkdir RESULTS_HTSEQ/FILTERED
#mkdir RESULTS_HTSEQ_REMOVED
#mkdir RESULTS_HTSEQ_REMOVED/NO_FILTERED
#mkdir RESULTS_HTSEQ_REMOVED/FILTERED

#####################################
######### CHECK STRANDENESS #########
## how_are_we_stranded_here v1.0.1 ##
#####################################

conda activate STRAND

# Done:

for Sample in E15_KO_NT E15_KO_P E15_KO_R24 E15_KO_R3 E15_WT_NT E15_WT_P E15_WT_R24 E15_WT_R3 E29_KO_NT E29_KO_P E29_KO_R24 E29_KO_R3 E29_WT_NT E29_WT_P E29_WT_R24 E29_WT_R3 E31_KO_NT E31_KO_P E31_KO_R24 E31_KO_R3 E31_WT_NT

do

	check_strandedness \
  	--gtf ./UBUNTU_LAB/DATA/INDEX/MOUSE/STAR/Genome_Mus_musculus_GrcM39_104/Mus_musculus.GRCm39.104.gtf \
  	--transcripts ./UBUNTU_LAB/DATA/INDEX/MOUSE/STAR/Genome_Mus_musculus_GrcM39_104/GRCm39_STAR_index/Mus_musculus.GRCm39.cdna.all.fa \
  	--reads_1 ./SAMPLES/NO_FILTERED/"$Sample"/"$Sample"_1.fq.gz \
  	--reads_2 ./SAMPLES/NO_FILTERED/"$Sample"/"$Sample"_2.fq.gz \
  	> ./STRAND/"$Sample".txt

done

#########################
##### Check quality #####
#########################
##### fqc 0.11.5 ########
#########################

# FastQC (v0.12.1) results previously generated by Novogene and verified locally.

# Done:

#for Sample in E15_KO_NT E15_KO_P E15_KO_R24 E15_KO_R3 E15_WT_NT E15_WT_P E15_WT_R24 E15_WT_R3 E29_KO_NT E29_KO_P E29_KO_R24 E29_KO_R3 E29_WT_NT E29_WT_P E29_WT_R24 E29_WT_R3 E31_KO_NT E31_KO_P E31_KO_R24 E31_KO_R3 E31_WT_NT

#do

#	fastqc ./SAMPLES/NO_FILTERED/"$Sample"_1.fastq.gz
#	fastqc ./SAMPLES/NO_FILTERED/"$Sample"_2.fastq.gz

#done

##############################
## Alignment (STAR 2.7.9a): ##
##############################

# Star_index GRCm39 --> ./UBUNTU_LAB/DATA/INDEX/MOUSE/STAR/Genome_Mus_musculus_GrcM39_104/GRCm39_STAR_index

# GTF GRCm39 --> ./UBUNTU_LAB/DATA/INDEX/MOUSE/STAR/Genome_Mus_musculus_GrcM39_104/Mus_musculus.GRCm39.104.gtf

######################
## Genome alignment ##
######################

	# --runMode: alignReads ==> Type of the run.
	# --clip3pAdapterSeq polyA
	# --genomeDir: ==> Directory where the genome is established.
	# --sjdbGTFfile ==> GTF file.
	# --outSAMtype BAM ==> BAM format (output).
	# SortedByCoordinate: output sorted by coordinate Aligned.sortedByCoord.out.bam file, similar to samtools sort 	command.
	# Inputs: 2 inputs files.

# FILTERED_NOVOGENE_VERSION:

# Done:

for Sample in E15_KO_NT E15_KO_P E15_KO_R24 E15_KO_R3 E15_WT_NT E15_WT_P E15_WT_R24 E15_WT_R3 E29_KO_NT E29_KO_P E29_KO_R24 E29_KO_R3 E29_WT_NT E29_WT_P E29_WT_R24 E29_WT_R3 E31_KO_NT E31_KO_P E31_KO_R24 E31_KO_R3 E31_WT_NT

do

	STAR --runMode alignReads --runThreadN 6 --readFilesCommand zcat --genomeDir ./UBUNTU_LAB/DATA/RNA-SEQ/INDEX/MOUSE/STAR/Genome_Mus_musculus_GrcM39_104/GRCm39_STAR_index --sjdbGTFfile ./UBUNTU_LAB/DATA/RNA-SEQ/INDEX/MOUSE/STAR/Genome_Mus_musculus_GrcM39_104/Mus_musculus.GRCm39.104.gtf --outSAMtype BAM SortedByCoordinate --outFileNamePrefix ./RESULTS_STAR/FILTERED_NOVOGENE/"$Sample"/"$Sample"_ --readFilesIn ./SAMPLES/FILTERED_NOVOGENE/"$Sample"/"$Sample"_1.clean.fq.gz ./SAMPLES/FILTERED_NOVOGENE/"$Sample"/"$Sample"_2.clean.fq.gz

done

# Using Novogene-filtered reads (*.clean.fq.gz), which have already had adapter/low-quality bases removed.

######################################################
## Count reads to features (Ensembl -> GRCm39.104): ##
######################################################

# -s no  --> library unstranded (confirmed by how_are_we_stranded_here)
# -a 10  --> minimum alignment quality (MAPQ)

conda activate htseq

# FILTERED_NOVOGENE_VERSION:

# Done:

for Sample in E15_KO_NT E15_KO_P E15_KO_R24 E15_KO_R3 E15_WT_NT E15_WT_P E15_WT_R24 E15_WT_R3 E29_KO_NT E29_KO_P E29_KO_R24 E29_KO_R3 E29_WT_NT E29_WT_P E29_WT_R24 E29_WT_R3 E31_KO_NT E31_KO_P E31_KO_R24 E31_KO_R3 E31_WT_NT

do

	htseq-count -a 10 -m intersection-nonempty -i gene_id -f bam -s no -r pos ./RESULTS_STAR/FILTERED_NOVOGENE/"$Sample"/"$Sample"_Aligned.sortedByCoord.out.bam ./UBUNTU_LAB/DATA/RNA-SEQ/INDEX/MOUSE/STAR/Genome_Mus_musculus_GrcM39_104/Mus_musculus.GRCm39.104.gtf > ./RESULTS_HTSEQ/FILTERED_NOVOGENE/"$Sample"/"$Sample"_htseq.txt

done

##############################################################################################################
